x-pack/plugins/apm/public/components/routing/templates/apm_main_template.tsx

```diff
@@ -49,9 +55,14 @@ export function ApmMainTemplate({
     hasData: data?.hasData,
   });
 
+  const byPassNoDataScreen = byPassNoDataScreenPaths.some((byPassPath) => {
+    const regex = new RegExp(byPassPath);
+    return regex.test(location.pathname);
+  });
```

Comment:
Since this is the main template and anything we add here could be rendered multiple times for every single page view we might want to be mindful about performance. Would the following suffice:

```suggestion
  const shouldBypassNoDataScreen = byPassNoDataScreenPaths.some(path => location.pathname.includes(path))
```

Reply:
... and then change:

```ts
const byPassNoDataScreenPaths = ['/app/apm/settings/'];
```